raiden_libs/network/matrix/client.py

```diff
@@ -175,6 +183,22 @@ def start_listener_thread(self, timeout_ms: int = 30000, exception_handler: Call
         self.should_listen = True
         self.sync_thread = gevent.spawn(self.listen_forever, timeout_ms, exception_handler)
 
+    def stop_listener_thread(self):
```

Comment:
If I understand correctly, the changes to this and the following function seem to be preparing a change in `raiden.network.transport.MatrixTransport`, where all greenlets are handled before calling `logout` (formerly `join_and_logout`) and the greenlets of `MatrixTransport`, which used to be passed to `join_and_logout` are handled explicitly there?

Why are we dropping the timeout when waiting for the greenlets?

Reply:
@jomuel the Timeout will be handled by the `MatrixTransport`, as it's the only one who cares for these timeouts, to keep compatibility with parent `logout` method.